### ISSUE TITLE
File names with plus (+) are registered but do not "exist"

### ISSUE THREAD

**Issue by shawnpyle** [mikey179/vfsStream#118]
It appears that file names with plus symbols in the file name fail to pass `file_exists` calls.

```php
$root = vfsStream::setup();
$this->assertFalse($root->hasChildren());
$this->assertFalse(file_exists(vfsStream::url('test.php')));
			
vfsStream::create(['test.php'=>'test']);
$this->assertTrue($root->hasChild('test.php'));
$this->assertTrue(file_exists('vfs://root/test.php'));

vfsStream::create(['test.html+haml.php'=>'test']);
$this->assertTrue($root->hasChild('test.html+haml.php'));
$this->assertTrue(file_exists('vfs://root/test.html+haml.php')); //=> FAILS
```

**Comment by mikey179** [comment]
In an URL the path must be urlencoded. This is due to #104. Thus, the correct vfs URL would be `vfs://root/test.html%2Bhaml.php`.

However, while investigating this I stumbled upon the issue that `vfsStream::path()` correctly urldecodes the path, but that `vfsStream::url()` never encodes it properly.

**Comment by mikey179** [comment]
I was wrong about this, please see #128. The code should work without url encoding the vfs url. Will be fixed in 1.6.2.
